### PR TITLE
[Concurrency] Add async versions of some functions on Sequence

### PR DESCRIPTION
**Pull request by drexin**
Resolves rdar://73627252

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by drexin** [comment]
@swift-ci smoke test linux

**Comment by drexin** [comment]
@swift-ci smoke test linux

**Comment by drexin** [comment]
@swift-ci smoke test windows

**Comment by drexin** [comment]
@swift-ci smoke test windows

**Comment by drexin** [comment]
@swift-ci smoke test linux

**Comment by drexin** [comment]
@swift-ci smoke test windows

**Comment by drexin** [comment]
@swift-ci smoke test windows

**Comment by drexin** [comment]
@swift-ci smoke test windows

**Comment by drexin** [comment]
@swift-ci smoke test windows

File: stdlib/public/Concurrency/SequenceExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      }
+    }
+    return nil
+  }
+
+  /// `async` version of `Sequence.prefix`.
+  ///
+  /// - SeeAlso: Sequence.prefix
+  @inlinable
+  public __consuming func prefix(
```
Comment (Catfish-Man) [inline stdlib/public/Concurrency/SequenceExtensions.swift:122]: Same question about consuming

**Comment by drexin** [comment]
@Catfish-Man I think all of your questions can be answered with: I just copied the non-async functions and added the `aync` to them 😅.

**Pull request closed without merging**